packages/scripts/src/utils/generate-loader.ts

```diff
@@ -0,0 +1,52 @@
+import type webpack from 'webpack';
+
+const resourceMatcher = /^\?generate-loader-(\w+)$/g;
+const virtualModules: Record<string, string> = {};
```

Comment:
I need the "virtual modules" I'm writing to be on some global state, since when the loader calls generate it seems to share the loader instance within the compiler, so I only get the data available to the first time I register the loader. 
Moving the virtualModules here gives me the global state I need